packages/pynumaflow-lite/manifests/reducestream/reducestream_counter.py

```diff
@@ -0,0 +1,108 @@
+"""
+Reduce Streaming Counter Example
+
+This example demonstrates how to use ReduceStreamer to emit intermediate results
+as data arrives, rather than waiting until all data is received.
+
+The counter increments for each datum and emits a message every 10 items,
+plus a final message at the end.
+"""
+import asyncio
+import signal
+from collections.abc import AsyncIterable, AsyncIterator
+
+from pynumaflow_lite import reducestreamer
+
+
+class ReduceCounter(reducestreamer.ReduceStreamer):
+    """
+    A reduce streaming counter that emits intermediate results.
+    
+    This demonstrates the key difference from regular Reducer:
+    - Regular Reducer: waits for all data, then returns Messages
+    - ReduceStreamer: yields Message objects incrementally as an async iterator
+    """
+    
+    def __init__(self, initial: int = 0) -> None:
+        self.counter = initial
+
+    async def handler(
+        self,
+        keys: list[str],
+        datums: AsyncIterable[reducestreamer.Datum],
+        md: reducestreamer.Metadata,
+    ) -> AsyncIterator[reducestreamer.Message]:
+        """
+        Process datums and yield messages incrementally.
+        
+        Args:
+            keys: List of keys for this window
+            datums: Async iterable of incoming data
+            md: Metadata containing window information
+            
+        Yields:
+            Message objects to send to the next vertex
+        """
+        iw = md.interval_window
+        print(f"Handler started for keys={keys}, window=[{iw.start}, {iw.end}]")
```

Comment:
nit: `log.info` instead of `print` ?

Reply:
i was using native entities so far in examples as much as possible